tractor/_broadcast.py

```diff
@@ -0,0 +1,201 @@
+'''
+``tokio`` style broadcast channel.
+https://tokio-rs.github.io/tokio/doc/tokio/sync/broadcast/index.html
+
+'''
+from __future__ import annotations
+from itertools import cycle
+from collections import deque
+from contextlib import contextmanager
+from functools import partial
+from typing import Optional
+
+import trio
+import tractor
+from trio.lowlevel import current_task
+from trio.abc import ReceiveChannel
+from trio._core._run import Task
+# from trio._channel import (
+#     MemoryReceiveChannel,
+# )
+
+
+class Lagged(trio.TooSlowError):
+    '''Subscribed consumer task was too slow'''
+
+
+class BroadcastReceiver(ReceiveChannel):
+    '''A memory receive channel broadcaster which is non-lossy for the
+    fastest consumer.
+
+    Additional consumer tasks can receive all produced values by registering
+    with ``.subscribe()``.
+
+    '''
+    def __init__(
+        self,
+
+        rx_chan: MemoryReceiveChannel,
```

Comment:
We should probably switch this to a `tri.abc.ReceiveChannel` right?

Since you should be able to pass in anything that duck-types that interface?